### PR TITLE
feat: git-hook 설정

### LINKED ISSUE #326: feat: git action -> git hook 기반으로 테스트 코드에 대해 검증을 수행한다
- 테스트가 실패하면 push가 불가능하도록 변경

### PR DESCRIPTION
**Pull request by vsh123**
resolve #326

### CODE REVIEW

File: git-hooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# !/bin/bash
+
+CURRENT_DIR=`pwd`
+PROJECT_ROOT="$( git rev-parse --show-toplevel )"
+
+cd $PROJECT_ROOT
+
+GRADLE_RESULT="$(./gradlew clean build)"
+
+exit $?
```
Comment (oeeen) [inline git-hooks/pre-push:10]: 👍

File: git-hooks/pre-push [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# !/bin/bash
+
+CURRENT_DIR=`pwd`
+PROJECT_ROOT="$( git rev-parse --show-toplevel )"
+
+cd $PROJECT_ROOT
+
+GRADLE_RESULT="$(./gradlew clean build)"
```
Comment (oeeen) [inline git-hooks/pre-push:8]: 💯

**Review by oeeen — APPROVED** [review]

File: git-hooks/setup.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# !/bin/bash
+PROJECT_ROOT=`git rev-parse --show-toplevel`
+rm ${PROJECT_ROOT}/.git/hooks/pre-push || true; ln -s ${PROJECT_ROOT}/git-hooks/pre-push ${PROJECT_ROOT}/.git/hooks/pre-push
```
Comment (oeeen) [inline git-hooks/setup.sh:3]: `command || true;` 👍

**Review by hyojaekim — APPROVED** [review]
good

**Pull request merged**